app/_src/gateway/key-concepts/routes/index.md

```diff
@@ -15,6 +15,96 @@ For example, if you have an external application and an internal application tha
 
 In the example above, two routes can be created, say `/external` and `/internal`, and both routes can point to `example_service`. A policy can be configured to limit how often the `/external` route is used and the route can be communicated to the external client for use. When the external client tries to access the service via {{site.base_gateway}} using `/external`, they are rate limited. But when the internal client accesses the service using {{site.base_gateway}} using `/internal`, the internal client will not be limited.
 
+## How {{site.base_gateway}} routes requests in `traditional_compat` mode
+
+For each incoming request, {{site.base_gateway}} needs to determine
+which service gets to handle it based on the routes that have been
+defined.  We describe how the routing process works in
+`traditional_compat` mode, in a logical sense, below.  The actual
+router implementation optimizes the process and caches results to
+avoid having to scan all routes repeatedly.  Note that when using the
+`expressions` router, route priorities are explicitly defined and this
+description does not apply.  Underneath, routes are converted from the
+`traditional_compat` format to the `expressions` format.
+
+In general, the router uses the highest priority matching route to
+handle a request.  If there are multiple matching routes with the same
+priority, it is not defined which of the two routes will be used.
```

Comment:
"Undefined" means that either of the routes will be chosen at the discretion of the implementation and we don't document which of the two it will be.  We used to use the "last change" timestamp of the route to enforce a particular ordering in the past, but as all those timestamps will be identical in hybrid mode or dbless, they are not useful as an additional sort criterion anymore.